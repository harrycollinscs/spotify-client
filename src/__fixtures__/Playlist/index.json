{
  "collaborative": false,
  "description": "",
  "external_urls": {
    "spotify": "https://open.spotify.com/playlist/3qQoF3PaRtg1iPUHKRzjEX"
  },
  "href": "https://api.spotify.com/v1/playlists/3qQoF3PaRtg1iPUHKRzjEX",
  "id": "3qQoF3PaRtg1iPUHKRzjEX",
  "images": [
    {
      "height": 640,
      "url": "https://mosaic.scdn.co/640/ab67616d0000b2732090f4f6cc406e6d3c306733ab67616d0000b27352c75ed37313b889447011efab67616d0000b27364442a4144ef18be67b37b37ab67616d0000b273ea9bbdf9f1433e68c7dffbad",
      "width": 640
    },
    {
      "height": 300,
      "url": "https://mosaic.scdn.co/300/ab67616d0000b2732090f4f6cc406e6d3c306733ab67616d0000b27352c75ed37313b889447011efab67616d0000b27364442a4144ef18be67b37b37ab67616d0000b273ea9bbdf9f1433e68c7dffbad",
      "width": 300
    },
    {
      "height": 60,
      "url": "https://mosaic.scdn.co/60/ab67616d0000b2732090f4f6cc406e6d3c306733ab67616d0000b27352c75ed37313b889447011efab67616d0000b27364442a4144ef18be67b37b37ab67616d0000b273ea9bbdf9f1433e68c7dffbad",
      "width": 60
    }
  ],
  "name": "ðŸš¿",
  "owner": {
    "display_name": "woodpush3r",
    "external_urls": {
      "spotify": "https://open.spotify.com/user/woodpush3r"
    },
    "href": "https://api.spotify.com/v1/users/woodpush3r",
    "id": "woodpush3r",
    "type": "user",
    "uri": "spotify:user:woodpush3r"
  },
  "primary_color": null,
  "public": true,
  "snapshot_id": "MzgsNDAxYjE0MTFkZjExOGQzMWQ2MTBkY2FiODY0NTRmNDZjY2E1ZjBjYQ==",
  "tracks": {
    "href": "https://api.spotify.com/v1/playlists/3qQoF3PaRtg1iPUHKRzjEX/tracks",
    "total": 11
  },
  "type": "playlist",
  "uri": "spotify:playlist:3qQoF3PaRtg1iPUHKRzjEX"
}
