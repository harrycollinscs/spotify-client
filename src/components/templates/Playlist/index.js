import axios from "axios";
import React, { PureComponent } from "react";
import styled from "styled-components";
import { base_api_url } from "../../../config";
import store from "../../../store";
import Page from "../../atoms/Page";
import Hero from "../../molecules/Hero";
import TrackList from "../../organisms/TrackList";

const ContentContainer = styled.div`
  padding: 30px 100px 30px 100px;
`;

const StyledTitle = styled.h2`
  font-weight: bold;
  padding-bottom: 20px;
`;

class Playlist extends PureComponent {
  state = {
    playlist: null,
  };

  getPlaylist(playlistId) {
    const playlistEndpoint = `${base_api_url}/playlists/${playlistId}`;

    axios
      .get(playlistEndpoint, {
        headers: {
          Authorization: "Bearer " + store.getState().accessToken,
        },
      })
      .then((res) => {
        console.log(res);
        this.setState({ playlist: res.data });
        console.log(this.state.playlist.tracks);
      });
  }

  componentDidMount() {
    const { id } = this.props.match.params;
    this.getPlaylist(id);
  }

  render() {
    const playlist = this.state.playlist;

    return (
      <Page>
        <Hero title={playlist?.name} content={playlist?.description} />
        <ContentContainer>
          <StyledTitle>Tracks</StyledTitle>
          {this.state.playlist ? (
            <TrackList tracks={playlist.tracks.items} />
          ) : (
            <p style={{ color: "black" }}>No tracks found</p>
          )}
        </ContentContainer>
      </Page>
    );
  }
}

export default Playlist;
