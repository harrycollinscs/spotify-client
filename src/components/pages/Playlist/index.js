import { connect } from 'react-redux';
import Playlist from '../../templates/Playlist';

const mapStateToProps = () => ({
});

const mapDispatchToProps = () => {
  
}

export default connect()(Playlist);
