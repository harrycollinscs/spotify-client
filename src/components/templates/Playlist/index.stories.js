import React from "react";
import Playlist from "./index";

export default {
  title: "Templates/Playlist",
  component: Playlist,
};

const Template = (args) => <Playlist {...args} />;

export const Default = Template.bind({});
Default.args = {
  match: {
    params: {
      id: "37i9dQZF1DWZixSclZdoFE",
    },
  },
};