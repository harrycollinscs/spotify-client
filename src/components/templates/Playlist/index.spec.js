import React from "react";
import Playlist from "./index";
import { shallow } from "enzyme";

describe("Templates/Playlist", () => {
  const props = {
    match: {
      params: {
        id: "37i9dQZF1DWZixSclZdoFE",
      },
    },
  };
  const component = <Playlist {...props} />;

  it("Should match snapshot", () => {
    const render = shallow(component);
    expect(render).toMatchSnapshot();
  });
});
