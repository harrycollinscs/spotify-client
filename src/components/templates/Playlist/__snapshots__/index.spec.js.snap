// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Templates/Playlist Should match snapshot 1`] = `
<Page
  darkBackground={false}
>
  <Hero
    withBackButton={true}
  />
  <styled.div>
    <styled.h2>
      Tracks
    </styled.h2>
    <p
      style={
        Object {
          "color": "black",
        }
      }
    >
      No tracks found
    </p>
  </styled.div>
</Page>
`;
